tests/python/pants_test/backend/python/rules/BUILD

```diff
@@ -3,7 +3,7 @@
 
 python_tests(
   name='inject_init',
-  sources=['test_inject_init.py'],
```

Comment:
My read on this affordance is that it supports legacy targets and the hope is for the irregularity to go away:
1. https://github.com/pantsbuild/pants/blob/ba0132d5409c0136d66d7427160601970cc2b1e3/src/python/pants/build_graph/target.py#L115-L120
2. source attribute is automatically promoted to sources #5908
3. Undocumented feature: https://www.pantsbuild.org/build_dictionary.html

My read could definitely be wrong though - @illicitonion seems like the expert here on intent.

Reply:
I'm largely indifferent to `source` vs `sources`; both work, there are guard-rails in place to stop you from using both, and I have no intention of trying to get rid of either...